kv/mem/store.go

```diff
@@ -183,14 +182,19 @@ func (s *store) CheckAndSet(key string, version int, val proto.Message) (int, er
 	}
 
 	newVersion := version + 1
+	s.updateInternal(key, newVersion, data)
+	return newVersion, nil
+}
+
+func (s *store) updateInternal(key string, newVersion int, data []byte) {
```

Comment:
nit: `updateInternalWithLock` might be better, it took me a few seconds to figure out this is called from within a `XXXWithLock` function.

Reply:
sg